src/execution_plan/ops/op_delete.c

```diff
@@ -10,55 +10,21 @@
 #include "../../util/qsort.h"
 #include <assert.h>
 
-#define nodeIDislt(a,b) (a<b)
-
-#define edgeIDislt(a,b) ( ( a->src < b->src ) ||\
-    ( ( a->src == b->src ) && ( a->dest < b->dest ) ) ||\
-    ( ( a->src == b->src ) && ( a->dest == b->dest ) && ( a->relation_type < b->relation_type ) ) )
-
-/* Forward declarations. */
-void _LocateEntities(OpDelete *op_delete, QueryGraph *graph, AST_DeleteNode *ast_delete_node);
-
-OpBase* NewDeleteOp(AST_DeleteNode *ast_delete_node, QueryGraph *qg, GraphContext *gc, ResultSet *result_set) {
-    OpDelete *op_delete = malloc(sizeof(OpDelete));
-
-    op_delete->gc = gc;
-    op_delete->qg = qg;
-    op_delete->node_count = 0;
-    op_delete->edge_count = 0;
-    op_delete->nodes_to_delete = malloc(sizeof(char*) * Vector_Size(ast_delete_node->graphEntities));
-    op_delete->edges_to_delete = malloc(sizeof(char*) * Vector_Size(ast_delete_node->graphEntities));
-    op_delete->deleted_nodes = array_new(Node, 32);
-    op_delete->deleted_edges = array_new(Edge, 32);
-    op_delete->result_set = result_set;
-    
-    _LocateEntities(op_delete, qg, ast_delete_node);
-
-    // Set our Op operations
-    OpBase_Init(&op_delete->op);
-    op_delete->op.name = "Delete";
-    op_delete->op.type = OPType_DELETE;
-    op_delete->op.consume = OpDeleteConsume;
-    op_delete->op.reset = OpDeleteReset;
-    op_delete->op.free = OpDeleteFree;
-
-    return (OpBase*)op_delete;
-}
-
 void _LocateEntities(OpDelete *op, QueryGraph *qg, AST_DeleteNode *ast_delete_node) {
+    AST *ast = AST_GetFromLTS();
     for(int i = 0; i < Vector_Size(ast_delete_node->graphEntities); i++) {
         char *entity_alias;
         Vector_Get(ast_delete_node->graphEntities, i, &entity_alias);
 
         // Current entity is a node.
+        int entityRecIdx = AST_GetAliasID(ast, entity_alias);
         Node *n = QueryGraph_GetNodeByAlias(qg, entity_alias);
```

Comment:
I'd like to introduce a faster type-checking mechanism than this at some point.

Reply:
checking for the entity type?